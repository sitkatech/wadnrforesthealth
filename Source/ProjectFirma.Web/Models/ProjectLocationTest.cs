using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using ProjectFirma.Web.UnitTestCommon;

namespace ProjectFirma.Web.Models
{
    [TestFixture]
    public class ProjectLocationTest : DatabaseDirectAccessTestFixtureBase
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void AreProjectLocationsNotBad()
        {
            const string sqlQueryString = @"
                        select  p.ProjectID,
                            p.ProjectName, 
                            p.ProjectLocationPoint.STDistance(l.ProjectLocationGeometry.STCentroid()) as Distance
                        from dbo.Project p
                        join (
                            SELECT l.ProjectID, geometry::UnionAggregate(l.ProjectLocationGeometry) as ProjectLocationGeometry
                            from dbo.vGeoServerProjectLocationDetailed l
                            group by l.ProjectID
                        ) l on p.ProjectID = l.ProjectID
                        where p.ProjectLocationPoint.STIntersects(l.ProjectLocationGeometry) = 0
                        and p.ProjectLocationPoint.STDistance(l.ProjectLocationGeometry.STCentroid())
                        > 0.001  
                        and p.CreateGisUploadAttemptID is not null";
            var result = ExecAdHocSql(sqlQueryString);
            Approvals.Verify(result.TableToHumanReadableString());
        }
        [TestFixtureSetUp]
        public void TestFixtureSetup()
        {
            BaseFixtureSetup();
        }

        [TestFixtureTearDown]
        public void TestFixtureTeardown()
        {
            BaseFixtureTeardown();
        }
    }
}