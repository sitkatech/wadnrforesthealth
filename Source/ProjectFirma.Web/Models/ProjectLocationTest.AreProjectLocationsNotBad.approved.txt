ProjectID	ProjectName	Distance
26940	2024-NES-0013	0.00138013081234313
24279	2023-NES-0825	0.0020423129181159
27051	2024-SES-0036	0.0973856501037123
27145	N/A	0.540895442939642
23172	2023-SES-0119	0.00628036102032228
27217	2024-SES-0054	0.00629083963062075
27378	2024-SES-0082	0.00272992639818756
23520	2023-SES-0205	0.00143922104900544
22074	93-100684	0.00473887903926327
23422	2023-SES-0172	0.0011927243656574
24272	2019-0428	0.0125245700286825
21905	2018-SES-0051	0.418218453148671